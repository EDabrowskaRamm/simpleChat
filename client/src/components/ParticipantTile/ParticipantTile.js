import React from 'react'

import { ParticipantTileRoot } from './ParticipantTile.shards'

const ParticipantTile = ({ name }) => <ParticipantTileRoot>
  {name}
</ParticipantTileRoot>

export default ParticipantTile
