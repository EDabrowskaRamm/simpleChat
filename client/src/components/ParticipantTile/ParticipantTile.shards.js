import styled from '@emotion/styled'

export const ParticipantTileRoot = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  height: 120px;
  padding: 0 40px;
  border-bottom: ${({ theme }) => theme.commons.borders.main};
`
