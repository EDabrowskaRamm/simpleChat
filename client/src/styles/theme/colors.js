const _white = '#FFF'
const _colorGrey56 = '#868c97'
const _colorGrey61 = '#979c9f'
const _colorGrey63 = '#9a9fa8'
const _colorGrey89 = '#e1e3e5'
const _colorGrey92 = '#ebebeb'

const _pictonBlue = '#3d90f1'
const _oxfordBlue = '#354052'

export const background = {
  primary: _white,
  secondary: _colorGrey92,
}

export const text = {
  primary: _oxfordBlue,
  secondary: _colorGrey63,
  link: _pictonBlue,
  placeholder: _colorGrey61,
  left: _colorGrey56,
}

export const border = {
  input: _colorGrey56,
  chat: _colorGrey89,
  transparent: 'transparent',
}
