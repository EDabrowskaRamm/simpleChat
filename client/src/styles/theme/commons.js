import { border } from './colors'

export const borders = {
  main: `1px solid ${border.chat}`,
  input: `1px solid ${border.input}`,
}
