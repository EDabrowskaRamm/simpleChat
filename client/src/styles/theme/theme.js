import * as colors from './colors'
import * as commons from './commons'

export default {
  colors,
  commons
}
