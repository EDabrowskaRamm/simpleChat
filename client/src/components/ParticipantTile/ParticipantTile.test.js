import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeProvider } from 'emotion-theming'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import ParticipantTile from './ParticipantTile'

import theme from '../../styles/theme/theme'
import initialState from '../../../__mocks__/fileMock'

expect.addSnapshotSerializer(serializer)

describe('renders ParticipantTile', () => {
  it('Snapshot', () => {
    const mockStore = configureStore()
    const mockData = initialState.participants.participantsList[0].name

    const component = renderer.create(
      <Provider store={mockStore(initialState)}>
        <ThemeProvider theme={theme}>
          <ParticipantTile name={mockData} />
        </ThemeProvider>
      </Provider>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
