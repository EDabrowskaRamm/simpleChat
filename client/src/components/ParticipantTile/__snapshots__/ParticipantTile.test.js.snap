// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ParticipantTile Snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 120px;
  padding: 0 40px;
  border-bottom: 1px solid #e1e3e5;
}

<div
  className="emotion-0"
>
  Jane Doe
</div>
`;
